src/sysusers/sysusers.c

```diff
@@ -567,6 +571,17 @@ static int write_temporary_shadow(const char *shadow_path, FILE **tmpfile, char
                         return errno ? -errno : -EIO;
         }
 
+        /* Append the remaining NIS entries if any */
+        while (sp) {
+                errno = 0;
+                if (putspent(sp, shadow) < 0)
+                        return errno ? -errno : -EIO;
```

Comment:
BTW I picked up the code above but I don't really see why we need to test if errno is not zero here. If putspent() returns -1 then errno must be set to, no ?